core/backend/src/rpc/tracing.ts

```diff
@@ -14,12 +14,15 @@ import { AsyncLocalStorage } from "async_hooks";
 import { BackendLoggerCategory } from "../BackendLoggerCategory";
 import { IModelHost } from "../IModelHost";
 
+/* eslint-disable deprecation/deprecation */
+
 /**
  * Utility for tracing Rpc activity processing. When multiple Rpc requests are being processed asynchronously, this
  * class can be used to correlate the current calltrace with the originating RpcActivity. This is used for automatic appending
  * of RpcActivity to log messages emitted during Rpc processing. It may also be used to retrieve the user accessToken
  * from the RpcActivity.
  * @public
+ * @deprecated in 3.6. The RPC system will be significantly refactored (or replaced) in version 5.0.
```

Comment:
Do we have any alternative today to using this or is it pre-emptively deprecating without replacement?

Reply:
I think this would only get potentially moved to a new package during the refactor. Seems like necessary functionality...

Reply:
Had an offline discussion with @swbsi and he is going to remove the deprecation from some of the critical-path public APIs that do not yet have a replacement. In the future once we have a replacement identified, they can be deprecated.

Reply:
I have removed RpcInterface, RpcActivity, RpcTrace, SerializedRpcActivity, and RpcResponseCacheControl from the deprecation list.
Everything else seemed internal to the RPC system (and ideally should never have been public anyway).

We could still do a pretty major refactor (or even replacement) even with these top-level classes undeprecated.
They would just just forward to the new system internally...

Reply:
sounds good, thanks!